commit 61a7c9df71ee4e6f94b56c20f0d37c6e17d5f284
Author: Florian Weimer <fweimer@redhat.com>
Date:   Mon Dec 2 14:53:16 2019 +0100

    elf/tst-dlopenfail: Disable --no-as-needed for tst-dlopenfailmod1.so
    
    Otherwise, the shared object dependency which triggers the load
    failure is dropped, invalidating the test.

diff --git a/elf/Makefile b/elf/Makefile
index bf7c41f38be42184..467e810e784bb96d 100644
--- a/elf/Makefile
+++ b/elf/Makefile
@@ -1543,8 +1543,11 @@ LDFLAGS-tst-finilazyfailmod.so = \
 $(objpfx)tst-dlopenfail: $(libdl)
 $(objpfx)tst-dlopenfail.out: \
   $(objpfx)tst-dlopenfailmod1.so $(objpfx)tst-dlopenfailmod2.so
-# Order matters here.  tst-dlopenfaillinkmod.so's soname ensures
-# a run-time loader failure.
+# Order matters here.  tst-dlopenfaillinkmod.so's soname ensures a
+# run-time loader failure.  --as-needed breaks this test because
+# nothing actually references tst-dlopenfailmod2.so (with its soname
+# tst-dlopenfail-missingmod.so).
+LDFLAGS-tst-dlopenfailmod1.so = -Wl,--no-as-needed
 $(objpfx)tst-dlopenfailmod1.so: \
   $(shared-thread-library) $(objpfx)tst-dlopenfaillinkmod.so
 LDFLAGS-tst-dlopenfaillinkmod.so = -Wl,-soname,tst-dlopenfail-missingmod.so
